runtime/compiler/z/codegen/J9BCDTreeEvaluator.cpp

```diff
@@ -5138,6 +5138,115 @@ J9::Z::TreeEvaluator::pdchkEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   TR_ASSERT_FATAL_WITH_NODE(node, bytesWithSpacesConst >= 0 && bytesWithSpacesConst < 32,
+      "Space bytes count(%d) for external decimal was outside expected range [0-31].\n",
+      bytesWithSpacesConst);
+
+   TR::DataType dataType = child->getDataType();
+   // Safe to downcast here because decimalLength can at most be precision + 1
+   int8_t decimalLength = static_cast<int8_t>(TR::DataType::getSizeFromBCDPrecision(dataType, precision));
+   TR_ASSERT_FATAL_WITH_NODE(node, decimalLength > 0 && decimalLength <= (static_cast<int8_t>(precision) + 1),
+      "External decimal length was expected to be greater than 0 and less than equal to precision + 1, "
+         "but decimal length was %d and precision was %d.\n",
+      decimalLength, precision);
+   TR_PseudoRegister *sourceReg = cg->evaluateBCDNode(child);
+   sourceReg = cg->privatizeBCDRegisterIfNeeded(node, child, sourceReg);
+   TR::MemoryReference *sourceMR = generateS390LeftAlignedMemoryReference(child, sourceReg->getStorageReference(), cg, decimalLength);
+
+   TR::Register *vZondedLowReg = cg->allocateRegister(TR_VRF);
+   TR::Register *vZondedHighReg = cg->allocateRegister(TR_VRF);
+
+   /**
+    * I3: / SSC LS DSC   STC DC
+    *     0 0   0  00000 000 00000
+    *    - Separate-Sign Control (SSC)  : 1 if sign is separate, otherwise 0 for embedded sign.
+    *    - Leading Sign (LS)            : 1 if sign is leading, otherwise 0 for trailing sign.
+    *    - Disallowed-Spaces Count (DSC): DC is total number of digits, DSC represents number of digits
+    *                                     with zone and digit format, rest can have space.
+    *                                     Only relevant when LS and SSC are 0.
+    *    - Sign-Test Control (STC)      : Specifies which codes are considered as valid sign codes.
+    *                                     110 (C,D,F) for embedded sign (will return false for non-preferred sign codes).
+    *                                     010 (4e,60) for sign separate (leading/trailing).
+    *                                     000 all sign codes are considered valid (hex) (A-F).
+    *    - Digits Count (DC)            : Integer specifying number of bytes to be verified. Does not include sign byte.
+    */
+   uint16_t zonedDecimalInfo = 0x0; // I3 operandgit d
+   uint8_t stc = 0x0;
+
+   int8_t dsc = static_cast<int8_t>(precision - bytesWithSpacesConst);
+
+   if (dataType == TR::ZonedDecimalSignTrailingSeparate) // Sign trailing separate
+      {
+      zonedDecimalInfo = 0x2; // set SSC bit
+      stc = 0x2;
+      dsc = 0x0;
+      }
+   else if (dataType == TR::ZonedDecimalSignLeadingEmbedded) // Sign leading embedded
+      {
+      zonedDecimalInfo = 0x1; // set LC bit
+      }
+   else if (dataType == TR::ZonedDecimalSignLeadingSeparate) // Sign leading separate
+      {
+      zonedDecimalInfo = 0x3; // set SSC and LS bits
+      stc = 0x2;
+      dsc = 0x0;
+      }
+   zonedDecimalInfo = (zonedDecimalInfo << 5) | dsc;
```

Comment:
I find this shifting to be confusing, as the values stored above i.e. SCC does not match the bit mapping or the instruction and comment above.  I think it's probably better for the SCC / LC bits to have the right values set - i.e. 0x40000000 / 0x2000000 (ideally as macros, so it's easy to read)... and the STC / DSC values can be OR'd in to the appropriate bit ranges.

Reply:
Fixed in https://github.com/eclipse-openj9/openj9/pull/20507/commits/01b9c78830f0f358730a6f011415c9637cdc6f33